engine/api/src/main/java/io/deephaven/engine/table/hierarchical/TreeTable.java

```diff
@@ -107,6 +107,15 @@ interface NodeOperationsRecorder extends
      */
     TreeTable withNodeOperations(@NotNull NodeOperationsRecorder nodeOperations);
 
+    /**
+     * Copy this TreeTable to the new table, replacing the source with newSource.
+     *
+     * @param newSource a new source table that must have the same definition as the source of this tree table
+     *
+     * @return The new TreeTable
+     */
+    TreeTable rebase(Table newSource);
```

Comment:
I proposed a different version for `RollupTable.rebase`. You should write something equivalent for trees. I wonder if a less-documented version belongs in the `HierarchicalTable` interface for ease-of-use in the server's implementation for the gRPC service.

Reply:
No gRPC, but I did update the comment.